core/src/test/scala/org/apache/spark/storage/DiskBlockObjectWriterSuite.scala

```diff
@@ -196,9 +204,81 @@ class DiskBlockObjectWriterSuite extends SparkFunSuite {
     for (i <- 1 to 500) {
       writer.write(i, i)
     }
+    val clazz: Class[_] = writer.getClass
+    val bsField: Field = clazz.getDeclaredField("bs")
+    bsField.setAccessible(true)
+    val bs = bsField.get(writer).asInstanceOf[OutputStreamWithCloseDetecting]
+
+    val objOutField: Field = clazz.getDeclaredField("objOut")
+    objOutField.setAccessible(true)
+    val objOut = objOutField.get(writer).asInstanceOf[SerializationStreamWithCloseDetecting]
+
```

Comment:
Instead of reflection, introduce a `private[storage]` method in `DiskBlockObjectWriter` for testing - to retrieve the specific fields.

Something like:
```
// For testing
private[storage] def getSerializerWrappedStream: InputStream = bs

// For testing
private[storage] def getSerializationStream: SerializationStream = objOut

```

Reply:
"Great suggestion, thanks a lot! I've made the changes."